src/main/java/com/battleship/backend/models/Game.java

```diff
@@ -57,9 +58,39 @@ public void removePlayerShip(Ship shipToRemove){
         playerShipList.remove(deletedShip);
     }
 
-    public void takeComputerTurn(){
+    public String takeComputerTurn(){
         Request hitRequest = computerPlayer.generateValidHitRequest();
         fire(getPlayerBoard(), hitRequest);
+        return shipHitResult(getPlayerBoard(), hitRequest);
+    }
+
+    public boolean isGameOver(){
+        return getPlayerBoard().isEveryShipSunk() || getComputerBoard().isEveryShipSunk();
+    }
+
+    public String getWinner(){
+        if(getPlayerBoard().isEveryShipSunk()){
+            return "Computer Player is the Winner!";
+        }else if(getComputerBoard().isEveryShipSunk()){
+            return "Human Player is the Winner!";
+        }else return "Game is not over";
+    }
+
+    //returns result of a hit request
+
+    public String shipHitResult(Boardable board, Request hitRequest){
+        String hitResult = "Missed!";
+        Sectionable hitSection = board.getSection(hitRequest.getRow(), hitRequest.getCol());
+         if(hitSection.getIsShip()){
+            hitResult = "The " + board.getName() + "'s " + hitSection.getShipName() + " has been hit!";
+            if(board.getShip(hitSection.getShipName()).checkIsSunk()){
+                hitResult +=  " The " + hitSection.getShipName() +" has been sunk!";
+            }
+         }
+        if(this.isGameOver()){
+            hitResult = "The Game is Over! " + this.getWinner();
```

Comment:
Non-blocking, but would it provide better user experience if the user was informed of which ship was sank prior to winning the game?

Reply:
Fixed.